### PR TITLE
Set up CI with Azure Pipelines

### PR DESCRIPTION
**Pull request by rainersigwald**
Starting the process of moving the CI build to Azure Pipelines.

### CODE REVIEW

**Comment by jeffkl** [comment]
For PR builds, this is the [appveyor.yml](https://github.com/Microsoft/MSBuildSdks/blob/master/appveyor.yml)

The only artifacts should be the the logs in case of failure.  There's no need for the NUPKGs from a PR build.

**Comment by MeikTranel** [comment]
> The only artifacts should be the the logs in case of failure.

Are we sure this is a good idea. As of right now this would publish binary logs, which contain basically everything. @KirillOsenkov taught me this is a big no-no. Builds happen on VSeng-MicroBuild.. are you guys sure there is no confidential environment variables?

**Comment by jeffkl** [comment]
That's an interesting question.  Do public users have access to build artifacts?  If possible, we could upload the logs but restrict access to them.

File: azure-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+variables:
+  SignType: 'Test'
+
+steps:
+- task: MSBuild@1
+  displayName: 'Build dirs.proj'
+  inputs:
+    solution: dirs.proj
+    platform: 'AnyCPU'
+    configuration: '$(Configuration)'
```
Comment (jeffkl) [inline azure-pipelines.yml]: Just set this to Debug

**Comment by rainersigwald** [comment]
Public users do have access to artifacts, but there should be no secrets available in the build environment. That's part of the reason we distinguish between `dnceng/public` and `dnceng/internal`.

File: azure-pipelines.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+pool:
+  name: Hosted VS2017
+
+variables:
+  SignType: 'Test'
```
Comment (jeffkl) [inline azure-pipelines.yml]: Actually I like that this is test signing, because I've broken signing before with a PR.

**Comment by MeikTranel** [comment]
@rainersigwald "dnceng/public and dnceng/internal." are these the azure devops orgs?

**Comment by rainersigwald** [comment]
@MeikTranel Yes. Builds that run under https://dnceng.visualstudio.com/public/_build shouldn't have access to any secrets--they should be "as though you ran it on your machine" but automated. For builds that need to sign, push to feeds, etc, there's a separate infrastructure. This is (lightly) documented here: https://github.com/dotnet/arcade/blob/master/Documentation/AzureDevOps/AzureDevOpsOnboarding.md.

**Pull request merged**